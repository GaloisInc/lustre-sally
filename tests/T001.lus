
type T = enum { A, B, C };
type S = T;

node f () returns (x : S)
let
  x = A;
  --%PROPERTY x = B;
tel


