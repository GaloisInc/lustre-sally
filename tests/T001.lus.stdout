[Lustre] Loading model from: "T001.lus"
Sally Model
===========

(define-state-type
   S
   (
     (Eq Bool)
     (x Int)
     (|gal-initializing| Bool))
   ())
(define-transition-system
   TS
   S
   (= |gal-initializing| true)
   (and
      (= next.|gal-initializing| false)
      (= next.x 0)
      (=
         next.Eq
         (= next.x 1))))

(query
   TS
   (or |gal-initializing| Eq))

[Lustre] Validating properties:
[Lustre] Property Prop on line 8...
  [Sally][Invalid]
[Trace] 
Prop on line 8:
 Step | |-> | x 
------+-----+---
 1    |     | A 

[Lustre] Summary:
  [Valid] 0
  [Unknown] 0
  [Invalid] 1
[Lustre] Model status: [Invalid]
